src/main/java/de/tum/in/www1/artemis/service/connectors/BambooService.java

```diff
@@ -213,9 +191,11 @@ public void configureBuildPlan(ProgrammingExerciseParticipation participation) {
      */
     @Override
     public void triggerBuild(ProgrammingExerciseParticipation participation) throws HttpException {
+        if (!isBuildPlanEnabled(participation.getBuildPlanId())) {
```

Comment:
we should avoid this and make sure that trigger is only invoked when the build plan is already enabled